### DISCUSSION TITLE
Making snapshot files dependent on a crate feature being enabled

### DISCUSSION THREAD

**Discussion by bradlarsen** [mitsuhiko/insta#566] [Q&A]
Is there any good way to make Insta's snapshot files dependent on a particular Cargo feature of my crate being enabled?

I have a number of tests that give different output when a particular feature `foo` is enabled or not. I still want to run the tests, but have them use differently-trained snapshot files depending on whether `foo` is enabled or not.

**Comment by max-sixty** [comment]
Would something like this work?

```rust
        #[cfg(feature = "foo")]
        let name = "with_foo";
        
        #[cfg(not(feature = "foo"))]
        let name = "without_foo";
        
        assert_snapshot!(name, output);
```

Not perfect, but possibly sufficient?

**Reply by bradlarsen** [reply]
I ended up doing something along these lines. Instead of changing the name based on the enabling of the `foo` feature, I duplicated the entire test function, giving it a new function name, and only one of them gets compiled.

Not pretty, but it's explicit, and it seems to work.
